File: models/custom_dashboard.py
from odoo import api, fields, models
class SchoolDashboardData(models.Model):
    _name = 'school.dashboard.data'
    _description = 'Dashboard Data'

    @api.model
    def get_dashboard_counts(self, institution_id=None, **kwargs):
        counts = {
            'student_count': 0,
            'teacher_count': 0,
            'class_count': 0,
            'section_count': 0,
        }
        if institution_id:
            domain = [('institution_id', '=', institution_id)]
            counts['student_count'] = self.env['school.student'].search_count(domain)
            counts['teacher_count'] = self.env['school.teacher'].search_count(domain)
            counts['class_count'] = self.env['school.class'].search_count(domain)
            counts['section_count'] = self.env['school.section'].search_count(domain)
        return counts

    @api.model
    def get_institutions(self, **kwargs):
        institutions = self.env['school.institution'].search([])
        return [{'id': inst.id, 'name': inst.name} for inst in institutions]

File: static/src/js/custom_dashboard.js
import { registry } from "@web/core/registry";
import { Component, useState, onWillStart } from "@odoo/owl";
import { rpc } from "@web/core/network/rpc";

class CustomDashboard extends Component {
    setup() {
        this.state = useState({
        institutions: [],
        selectedInstitution: null,
        studentCount: 0,
        teacherCount: 0,
        classCount: 0,
        sectionCount: 0,});

    onWillStart(async () => {
        // Load institutions
        this.state.institutions = await rpc("/web/dataset/call_kw", {
        model: "school.dashboard.data",
        method: "get_institutions",
        args: [],
        kwargs: {},});});
    }
    async onInstitutionChange(ev) {
            const institutionId = parseInt(ev.target.value);
            this.state.selectedInstitution = institutionId || null;

            const counts = await rpc("/web/dataset/call_kw", {
            model: "school.dashboard.data",
            method: "get_dashboard_counts",
            args: [],
            kwargs: { institution_id: institutionId },
        });

        this.state.studentCount = counts.student_count;
        this.state.teacherCount = counts.teacher_count;
        this.state.classCount = counts.class_count;
        this.state.sectionCount = counts.section_count;
    }

}

CustomDashboard.template = "CustomDashboardTemplate";
CustomDashboard.props = {};
registry.category("actions").add("custom_dashboard", CustomDashboard);
